migrate/labels.yml

```diff
@@ -0,0 +1,122 @@
+# Repository labels to be synced across the entire org.
+# Every item must have a name, color, and description.
+
+# Notes:
+# * Repositories may define their own labels as long as they don't
+#   conflict with what's listed here. If it turns out a repository-specific
+#   label would be useful to have in the entire organization, we can always add
+#   it here post facto.
+# * We have added these based on need. They are not all-encompassing. If something
+#   seems missing, feel free to open a PR.
+# * Whenever a repo has a matching label, that label will be updated. If there
+#   is no matching label, the label will be created anew. "Matching" is determined
+#   by comparing the label's lowercased name, with all puncutation, emoji, and special
+#   chars removed.
+# * For picking hex colors, you might use: https://www.htmlcolor-picker.com/
+#   Describe your color with a comment so it's easier to review.
+
+
+### LABELS INDICATING HOW ISSUES CAN BE ENAGAGED WITH.
+
+- name: "good first issue"
+  color: "42dd35"  # friendly lime green
+  description: "A good task for a newcomer to start with"
+
+- name: "help wanted"
+  color: "54976d"  # fenway green
+  description: "Ready to be picked up by anyone in the community"
+
+
+### LABELS INDICATING BROAD THEMES OF WORK.
+### MORE THAN ONE OF THESE MAY APPLY AT A TIME.
+
+- name: "maintenance"
+  color: "ff9125"  # construction orange
+  description: "Relates to platform health and/or routine upkeep"
```

Comment:
Would code cleanup/refactoring fit under this? I feel like codebase health fits under platform health, but I'm curious as to how others interpret this label.

Reply:
Good question, I'm torn it this. When I think of `maintenance` I think moreso of upgrades or other keeping-the-lights-on work, rather than proactive improvements. Do we want a separate label like `refactoring`, `platform-health`, or `technical-investment` ? (or `tech-debt`, but I've come to feel that the term has developed an unfortunate connotation of "let's deprioritize this").

Reply:
Of those suggestions I like `refactoring` and `technical-investment` best. I'm not sure about `platform-health`, I'm not sure what would make me like it more, maybe `codebase-health`?

Overall I think I'm leaning towards `refactoring`.

Reply:
👍🏻

Reply:
I keep coming back to "who will apply the label" and "who will read the label".  I don't understand what the separate labels like "refactoring" will do? How will they help people find issues/PRs?

Reply:
> I keep coming back to "who will apply the label" and "who will read the label". I don't understand what the separate labels like "refactoring" will do? How will they help people find issues/PRs?

I guess my thought on the split between `maintenance` and `refactoring` is `maintenance` implies "if we don't do this things will break," where as `refactoring` implies "this will be nice for making the codebase easier to work in."

With that in mind, I would want to make sure that any `maintenance` PRs were reviewed/merged quite quickly, where as `refactoring` PRs could stay in the review state and gather feedback for a bit longer.

Reply:
Wouldn't the `refactoring` label be doing the same job as the `refactor:` conventional commit category?

Reply:
Sure, much like `bug` is doing the same job as `fix:`, and `enhancement` is doing the same job as `feature:`, and `documentation` is doing the same job as `docs:`.

Unresolved issues don't have commits, though, so we can't use conventional commit types to label them.

Perhaps each conventional commit type should map to a label? @nedbat , what do you think?

Reply:
I'd expect a lot of `maintenance` PRs to have commits that go under `chore:`. I think my main point is I'd like to be able to "filter by label" and see how many PRs need somewhat immediate attention because they have the `maintenance` label without needing to look at additional information.

I think in general I'm less hoping to have a `refactor` label than hoping to not have refactoring PRs clutter up the `maintenance` label.

Reply:
After thinking about this, I ended up landing on `code health`.  I think that without a separate label, this stuff would end up lumped in with `maintenance`.

I almost went with `refactoring`, but Ned's comment got me thinking that we should avoid just re-implementing conventional commits here. The (imagined) purpose of these labels is to broadly categorize issues and PRs, and both `maintenance` and `code health` PRs are likely to have a mix of `feat`, `fix`, `refactor`, and `chore` commits depending on exactly what type of code change is required.